openedx/core/djangoapps/credit/views.py

```diff
@@ -29,6 +29,61 @@
 log = logging.getLogger(__name__)
 
 
+@require_GET
+def get_providers_detail(request):
+    """
+
+    **User Cases**
+
+        Returns details of the credit providers filtered by provided query parameters.
+
+    **Parameters:**
+
+        * provider_id (list of provider ids separated with ","): The identifiers for the providers for which
+        user requested
+
+    **Example Usage:**
+
+        GET /api/credit/v1/providers?provider_id=asu,hogwarts
+        "response": [
+            "id": "hogwarts",
+            "display_name": "Hogwarts School of Witchcraft and Wizardry",
+            "url": "https://credit.example.com/",
+            "status_url": "https://credit.example.com/status/",
+            "description": "A new model for the Witchcraft and Wizardry School System.",
+            "enable_integration": false,
+            "fulfillment_instructions": "
+            <p>In order to fulfill credit, Hogwarts School of Witchcraft and Wizardry requires learners to:</p>
+            <ul>
+            <li>Sample instruction abc</li>
+            <li>Sample instruction xyz</li>
+            </ul>",
+        },
+        ...
+        ]
+
+    **Responses:**
+
+        * 200 OK: The request was created successfully. Returned content
+            is a JSON-encoded dictionary describing what the client should
+            send to the credit provider.
+
+        * 404 Not Found: The provider does not exist.
+
+    """
+    provider_id = request.GET.get("provider_id", None)
+    if not provider_id:
+
+        # If provider_id is empty or None then empty dictionary will be sent to user,
+        # Because "api.get_credit_providers()" may be called from other views for
+        # list of all credit providers
```

Comment:
I'd prefer to return the list of all providers if no `provider_id` parameter is specified for consistency with our other APIs.